FuelLabs/sway
# forc template implementation

closes #1380 

This PR introduces `forc template` and removes `forc init --template`. `forc template` currently takes the following inputs:

- `--url` : git repo URL
- `--project-name` : name of the project
- ` --template-name` : (optional) if specified provided name is searched in the repo. (for example `counter` example can be searched in [sway repo](https://github.com/FuelLabs/sway))

The current implementation fetches the HEAD of the default branch of the repo URL. 

**Example Usage** 

`forc template --project-name test_project --url https://github.com/FuelLabs/sway --template-name counter`

`forc template --project-name test_project --url https://github.com/owner/repo`

---

**Testing Steps**

`cargo run --bin forc template --project-name test_project --url https://github.com/FuelLabs/sway --template-name counter` : should result in a new folder created "test_project" with `counter` example inside it. Also, the forc.toml will be edited.

`cargo run --bin forc template --project-name test_project --url https://github.com/FuelLabs/sway` : should result in an error since there is no `Forc.toml` in the root. (Because there is no `Forc.toml` in the root and no template name is provided which means the template should be in the root directory).

I am leaving this as a draft since there might be some other desired behavior.
